test/utils/matchers.go

```diff
@@ -167,12 +167,16 @@ func WithOwnerReferences(matcher gtypes.GomegaMatcher) gtypes.GomegaMatcher {
 	}, matcher)
 }
 
-// WithPodTemplateAnnotations returns the deployments PodTemplate's annotations
+// WithPodTemplateAnnotations returns the PodTemplate's annotations
 func WithPodTemplateAnnotations(matcher gtypes.GomegaMatcher) gtypes.GomegaMatcher {
 	return gomega.WithTransform(func(obj Object) map[string]string {
 		dep, ok := obj.(*appsv1.Deployment)
```

Comment:
Could this not cast to the `podController` interface? Would get rid of the need for this double `if-else`?

Reply:
Or better still just make the anonymous function accept a `podController` instead of `Object` and then the tests will panic by themselves if the incorrect interface is passed

Reply:
podController is defined as a private interface. Do you want it to be public so it can be used in this way? We may have to refactor the tests to send a pod controller instead of `deployment` and `statefulset` types.

Reply:
Yeah, I think that would probably be better to do. Let's make `PodController` public and I think the tests should just work anyway? As in I think we are already passing `PodController`s in in most places. Lets give it a go and see what happens?

Reply:
Are you thinking something like this?
```
func WithPodTemplateAnnotations(matcher gtypes.GomegaMatcher) gtypes.GomegaMatcher {
	return gomega.WithTransform(func(obj core.PodController) map[string]string {
		return obj.GetPodTemplate().GetAnnotations()
	}, matcher)
}
```

If so
```
Error: Transform function expects 'core.PodController' but we have '*v1.Deployment'
```
happens on about ~20 times when the `eventually` function uses `utils.WithPodTemplateAnnotations(HaveKey(core.ConfigHashAnnotation))`

Reply:
Ok, I've had a deeper look at this, definitely not worth trying to use `podController` here, it's going to cause so many issues

Instead can we make it a switch as below? I think it will look cleaner, WDYT?

```
return gomega.WithTransform(func(obj Object) map[string]string {
		switch obj.(type) {
		case *appsv1.Deployment:
			return obj.(*appsv1.Deployment).Spec.Template.GetAnnotations()
		case *appsv1.StatefulSet:
			return obj.(*appsv1.StatefulSet).Spec.Template.GetAnnotations()
		default:
			panic("unknown pod template type")
		}
	}, matcher)
```

Reply:
Yeah, thats much easier to read. I've updated the PR.